### PR TITLE
Improve BriefcaseManager.releaseBriefcase logging

### PR DESCRIPTION
**Pull request by paulius-valiunas**
Currently, if anything fails in `releaseBriefcase`, it just logs an error with no information except iModel id and briefcase id, and returns silently. I think the errors should either (a) be handled in some way or (b) be thrown to the caller, or (c) it should at least provide more information about the error so we can draw some conclusions from the log. I don't see a good way to handle errors in this case and (b) would break the existing expectation (this method never throws), so I went with (c).

### CODE REVIEW

File: core/backend/src/BriefcaseManager.ts [hunk 1]
```diff
@@ -318,6 +318,7 @@ export class BriefcaseManager {
       await IModelHost.iModelClient.briefcases.get(requestContext, iModelId, new BriefcaseQuery().byId(briefcaseId));
       requestContext.enter();
     } catch (err) {
+      Logger.logInfo(loggerCategory, "Could not find briefcase to release", () => ({ iModelId, briefcaseId }));
```
Comment (kabentley) [inline core/backend/src/BriefcaseManager.ts:321]: @paulius-valiunas i would just take out the try/catch and let callers deal with this. How did they get the invalid ids to begin with?

Comment (paulius-valiunas) [reply core/backend/src/BriefcaseManager.ts:321]: I didn't get invalid ids, but there were cases where the tile agent didn't release its briefcase for unknown reasons. I got the log message from line 332 and it was totally useless.

I wanted to just remove the try/catches in this method, but I thought someone might have a reason to have it this way. I know you rewrote a big part of this class recently, so if you think it's fine to just throw these errors, I'll be happy to remove them.

Comment (kabentley) [reply core/backend/src/BriefcaseManager.ts:321]: I didn't change anything in this code, but i don't think it's right for it to swallow errors. Nobody can be relying on that.

Comment (paulius-valiunas) [reply core/backend/src/BriefcaseManager.ts]: yeah that makes sense, I just wanted to hear a second opinion. I changed it to throw these errors but kept one of the catch blocks to add that log message. That should make it easier to see which of the two requests failed.

**Review by kabentley — APPROVED** [review]

**Comment by paulius-valiunas** [comment]
@Mergifyio backport release/2.13.x

**Pull request merged**